sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -70,12 +70,28 @@ class SparkSqlAstBuilder extends AstBuilder {
 
   /**
    * Create a [[ShowTablesCommand]] logical plan.
+   * Example SQL :
+   * SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards']
    */
   override def visitShowTables(ctx: ShowTablesContext): LogicalPlan = withOrigin(ctx) {
-    if (ctx.LIKE != null) {
-      logWarning("SHOW TABLES LIKE option is ignored.")
+    if (ctx.qualifiedName() != null) {
+      logWarning("SHOW TABLES qualifiedName is ignored.")
```

Comment:
The only reason for qualifiedName to be used was Hive. If Hive doesn't support it  (it is supported in the parser BTW), then we should drop it.